code/modules/xenomorph/xeno_structures.dm

```diff
@@ -0,0 +1,110 @@
+		if(3)
+			take_damage(70)
+
+/obj/structure/resin/attack_animal(mob/living/user)
+	return
+
+/obj/structure/resin/attack_paw(mob/user)
+	return
+
+/obj/structure/resin/attack_larva(mob/living/carbon/xenomorph/larva/larva_attacker)
+	return
+
+/obj/structure/resin/attack_alien(mob/living/carbon/xenomorph/xeno_attacker)
+	return
+
+/obj/structure/resin/attack_hand(mob/living/user)
+	to_chat(user, "<span class='warning'>You scrape ineffectively at \the [src].</span>")
+	return TRUE
+
+/obj/structure/resin/bullet_act(obj/item/projectile/proj)
+	take_damage(proj.damage * 0.5)
+	return TRUE
+
+/obj/structure/resin/attackby(obj/item/hitting_item, mob/user, params)
+	if(hitting_item.flags_item & NOBLUDGEON)
+		return attack_hand(user)
+	. = ..()
+	user.do_attack_animation(src)
+	playsound(src, "alien_resin_break", 25)
+	take_damage(hitting_item.force * 0.5)
+
+/obj/structure/resin/flamer_fire_act()
+	take_damage(10)
+
+/obj/structure/resin/fire_act()
+	take_damage(10)
+
+
+/obj/structure/resin/silo
+	icon = 'icons/Cimex/3x3structures.dmi'
+	icon_state = "silo_dark"
+	name = "resin silo"
+	desc = "A slimy, oozy resin bed filled with foul-looking egg-like ...things."
+	bound_width = 96
+	bound_height = 96
+	max_integrity = 400
+	var/turf/center_turf
+	var/datum/hive_status/associated_hive
+	var/silo_area
+
+
+/obj/structure/resin/silo/Initialize()
+	. = ..()
+	GLOB.xeno_resin_silos += src
+	center_turf = get_step(src, NORTHEAST)
+	if(!istype(center_turf))
+		center_turf = loc
+	return INITIALIZE_HINT_LATELOAD
+
+
+/obj/structure/resin/silo/LateInitialize()
+	. = ..()
+	if(!locate(/obj/effect/alien/weeds) in center_turf)
+		new /obj/effect/alien/weeds/node(center_turf)
+	associated_hive = GLOB.hive_datums[XENO_HIVE_NORMAL]
+	if(associated_hive)
+		RegisterSignal(associated_hive, list(COMSIG_HIVE_XENO_MOTHER_PRE_CHECK, COMSIG_HIVE_XENO_MOTHER_CHECK), .proc/is_burrowed_larva_host)
+	silo_area = get_area(src)
+
+
+/obj/structure/resin/silo/Destroy()
+	. = ..()
+	GLOB.xeno_resin_silos -= src
+
+
+/obj/structure/resin/silo/structure_destruction()
+	if(associated_hive)
+		associated_hive.xeno_message("<span class='xenoannounce'>A resin silo has been destroyed at [silo_area]!</span>", 2, TRUE)
+	playsound(src, "sound/effects/alien_egg_burst.ogg", 25)
+	qdel(src)
+
+
+/obj/structure/resin/silo/examine(mob/user)
+	. = ..()
+	switch(obj_integrity)
+		if(350 to 400)
+			to_chat(user, "<span class='info'>It appears in good shape, pulsating healthiy.</span>")
+		if(250 to 350)
+			to_chat(user, "<span class='warning'>It's slightly damaged, but still seems healthy.</span>")
+		if(150 to 250)
+			to_chat(user, "<span class='warning'>It's quite beat up, but it seems alive.</span>")
+		if(50 to 150)
+			to_chat(user, "<span class='warning'>It looks severely damaged, its movements slow.</span>")
+		if(0 to 50)
+			to_chat(user, "<span class='warning'>It's barely holding, there's leaking oozes all around, and most eggs are broken. Yet it is not inert.</span>")

```

Comment:
note for the future, doing switches like this breaks if you ever updated the `obj_integrity`, its better to get a HP % and use that in the switch.

Reply:
Agreed.